core/block/source/source_test.go

```diff
@@ -42,3 +56,108 @@ func Test_snapshotChance2(t *testing.T) {
 	// here is an example of distribution histogram
 	// https://docs.google.com/spreadsheets/d/1xgH7fUxno5Rm-0VEaSD4LsTHeGeUXQFmHsOm29M6paI
 }
+
+func TestSource_PushChange(t *testing.T) {
```

Comment:
If you extract checking the size of snapshot into a separate function, you don't need such complicated testing code

Reply:
Thank you for advice! It simplifies drastically